continuedev/continue
# feat: add Quick Actions CodeLens feature

## Description

Adds a new "Quick Actions" affordance. Quick Actions are VSCode Code Lens that allow users to perform common actions on functions and classes, such as explaining how the code behaves or generating a docstring.

## Checklist


## Screenshots

![quick-actions-demo-small](https://github.com/continuedev/continue/assets/20157849/9927a3dc-9384-45ff-b05e-8e6be1cec1e8)

## Testing

- Try out both default actions
- Disable the feature in settings, confirm the code lens is removed
- Reenable it, confirm is shows back up
- Try adding a custom action, e.g. 
```json
"experimental": {
    "quickActions": [
      {
        "title": "Detailed explanation",
        "prompt": "Explain the following code in detail, inlcuding all methods and properties.",
        "sendToChat": true
      }
    ]
  }
```
- Confirm the custom command works
